tests/__init__.py

```diff
@@ -35,3 +42,41 @@ def unreadable_pdf(tmp_path: Path) -> str:
     file_path = tmp_path / "document.pdf"
     file_path.touch(mode=0o000)
     return str(file_path)
+
+
+def get_qt_app() -> Application:
+    if Application.instance() is None:
+        return Application()
+    else:
+        return Application.instance()
+
+
+def generate_isolated_updater(
+    tmp_path: Path, monkeypatch: MonkeyPatch, enable_qt: bool = False
+) -> Updater:
+    """Generate an Updater class with its own settings."""
+    if enable_qt:
+        app = get_qt_app()
+    else:
+        app = mock.MagicMock()
```

Comment:
Are you referring to the Dangerzone `Application` or the `Updater` object? The former is indeed a fixture:

https://github.com/freedomofpress/dangerzone/blob/e6cd583946e26d09c829afd286b4b2ce3a8f9f48/tests/__init__.py#L80-L82

The latter is a bit more complicated, as it's a Qt singleton that I feel kind of iffy when we instantiate it. But I can turn it into a fixture, if you think it's worth it.

Reply:
Sorry for not making it clear. I mean instead of importing the `mock` module, using the mocker fixture.

Reply:
Oh right. It's because this function is not a fixture, due to the `enable_qt` flag. If you want to, I can make the mocker object an argument of this function, and make the callers pass a mocker fixture.

Reply:
Yes. I think that would be preferable.